### PR TITLE
Update to Angular 6

### PR DESCRIPTION
**Pull request by realModusOperandi**
Once merged, this PR might cause random issues. If you are seeing an issue, you can try `./gradlew cleanNpm` followed by the standard build to cause npm to reinstall all packages.

This changes a few minor things, including the way rxjs imports are done. Unfortunately due to some bug I did have to pull in the old-style module for one import. The Angular project is configured in angular.json instead of .angular-cli.json. Animations are no longer polyfilled so the app is a whole 47KB smaller!!!

### CODE REVIEW

File: frontend/prebuild/package.json [hunk 1]
```diff
@@ -13,48 +13,49 @@
...
     "karma-cli": "~1.0.1",
     "karma-coverage-istanbul-reporter": "^1.4.2",
-    "karma-jasmine": "~1.1.0",
+    "karma-jasmine": "^1.1.2",
     "karma-jasmine-html-reporter": "^0.2.2",
     "protractor": "~5.1.2",
     "ts-node": "~3.2.0",
     "tslint": "~5.9.1",
-    "typescript": "~2.5.3"
+    "typescript": ">=2.7.2 <2.8.0"
```
Comment (aguibert) [inline frontend/prebuild/package.json]: why downgrading typescript?

File: frontend/prebuild/src/app/game/game.service.ts [hunk 2]
```diff
@@ -18,20 +18,20 @@ export class GameService {
 
     socketService.url = `${environment.API_URL_GAME_WS}/${this.roundId}`;
     this.messages = <Subject<Object>>socketService.socket
-    .map((response: MessageEvent): any => {
-      //console.log(`Game service handling message: ${response.data}`);
+    .pipe(map((response: MessageEvent): any => {
+      console.log(`Game service handling message: ${response.data}`);
```
Comment (aguibert) [inline frontend/prebuild/src/app/game/game.service.ts]: can we leave this log statement commented out?  It fills up logs very fast

File: frontend/prebuild/package.json (hunk 1 above)
Comment (realModusOperandi) [reply frontend/prebuild/package.json]: How do you mean? 2.7 is newer...

File: frontend/prebuild/src/app/game/game.service.ts (hunk 2 above)
Comment (realModusOperandi) [reply frontend/prebuild/src/app/game/game.service.ts]: Don’t know how it got put back in. Must have been a bad merge. I can fix it.

File: frontend/prebuild/package.json (hunk 1 above)
Comment (aguibert) [reply frontend/prebuild/package.json]: oops, managed to read the diff backwards...

**Pull request merged**